Add a document describing Redis storage keys

mayorova opened the pull request:
Adding a document that describes the keys stored in Backend Redis (storage, not queues, because the latter is managed by resque).

jlledom reviewed (approved):
Thanks for doing this!! 🙏🙏

mayorova commented:
> Wow, this is amazing work!

Thank you! :heart: 

> I would just suggest to mention that the two kinds of data are stored in separate Redis databases.

That's not actually true. All of this data is stored in the same DB. I kind of mentioned that this document is talking about "storage", which is one DB. The other DB is called "queues", and it is managed by resque. I don't think we have anything "custom" there, but I'll double-check.

> Also would be nice to mention somewhere that the statistical data is generated based on API calls from apicast as well I think we had something on porta side that was sending stats about providers accessing porta.

Well, actually, backend doesn't really care where the requests come from, the stats data are generated when usage is reported through `authrep.xml` or `transactions.xml` calls, it can be through APIcast, directly to the API (e.g. through ActiveDocs in the admin portal), or through some client - like in the case of `ReportTrafficWorker` in porta, which is what I guess you were referring to.

I can add a note in the doc to clarify that.

mayorova commented on docs/redis_keys.md:
```diff
@@ -39,14 +40,15 @@ The keys in the second category are the ones generated by _backend_ itself. Thes
+| `application/service_id:SERVICE_ID/id:APP_ID/plan_name`        | SERVICE_ID, APP_ID   | string | `application/service_id:2/id:593a8c1a/plan_name`                   | `"Test plan"`                                | Getting the plan name of the application by its app ID and service ID                            |
+| `application/service_id:SERVICE_ID/id:APP_ID/state`            | SERVICE_ID, APP_ID   | string | `application/service_id:2/id:593a8c1a/state`                       | `"active"`, or `"pending"`, or `"suspended"` | Getting the state of the application by its app ID and service ID                                |
+| `application/service_id:SERVICE_ID/id:APP_ID/keys`             | SERVICE_ID, APP_ID   | set    | `application/service_id:2/id:593a8c1a/keys`                        | `1) "91efe8a356c83e94b40e54cce211a1a4"`      | List of application keys associated to the application (for App Id / App Key mode)               |
+| `application/service_id:SERVICE_ID/id:APP_ID/redirect_url`     | SERVICE_ID, APP_ID   | string | `application/service_id:2/id:593a8c1a/redirect_url`                | `"https://example.com/redirect"`             | Redirect URL of the aplication (for application of services with OAuth/OpenID Connect auth mode) |
```
I just added this line in the 2nd commit, the diff doesn't show it clearly because of reordering and fixing the columns widths.

mayorova commented:
@akostadinov hopefully I addressed your comments here: https://github.com/3scale/apisonator/pull/424/commits/57c581936d95a22526fec68742f154de5ba0b30b

mayorova commented on docs/redis_keys.md:
```diff
@@ -0,0 +1,201 @@
+### Stats
+
+Stats keys hold the counters for application usage, reported through the Service Management API, specifically, `/transactions/authrep.xml`, `transactions/oauth_authrep.xml`, and `/transactions.xml` endpoints. For example, sending `usage[hits]=1` in authrep calls or `transactions[0][usage][hits]=1` in transaction calls will increase the counter for the metric "hits" by `1`.
+
```
I expanded the explanation and pushed it in https://github.com/3scale/apisonator/pull/424/commits/0322a74c4dcc042aca0e33f361a42fe624aa8d97

mayorova replied:
I also saw some issues with reporting Analytics API and opened a Jira: https://issues.redhat.com/browse/THREESCALE-11812

akostadinov replied:
Wow, good catch!
